contracts/Core/StakeManager.sol

```diff
@@ -173,44 +176,61 @@ contract StakeManager is Initializable, StakeStorage, StateManager, Pause, Stake
     /// @param stakerId The Id of staker associated with sRZR which user want to unstake
     /// @param sAmount The Amount in sRZR
     function unstake(uint32 stakerId, uint256 sAmount) external initialized whenNotPaused {
+        Structs.Staker storage staker = stakers[stakerId];
+        uint32 epoch = _getEpoch(epochLength);
+        require(staker.id != 0, "staker.id = 0");
+        require(staker.stake > 0, "Nonpositive stake");
+        require(locks[msg.sender][staker.tokenAddress][LockType.Unstake].amount == 0, "Existing Unstake Lock");
+        require(locks[msg.sender][staker.tokenAddress][LockType.Withdraw].amount == 0, "Existing Withdraw Lock");
+        require(sAmount > 0, "Non-Positive Amount");
+
+        IStakedToken sToken = IStakedToken(staker.tokenAddress);
+        require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
+
+        locks[msg.sender][staker.tokenAddress][LockType.Unstake] = Structs.Lock(sAmount, 0, epoch + lockPeriod);
```

Comment:
You actually need to transfer the stoken from user. It's not really a "lock" if user just sells the srzr after locking. defeats the purpose and doesnt prevent attacks.